maas/resource_maas_volume_group.go

```diff
@@ -0,0 +1,259 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strconv"
+	"strings"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceMaasVolumeGroup() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a resource to manage MAAS Volume Groups, and construct them from partion-less block devices.",
+		CreateContext: resourceMaasVolumeGroupCreate,
+		ReadContext:   resourceMaasVolumeGroupRead,
+		UpdateContext: resourceMaasVolumeGroupUpdate,
+		DeleteContext: resourceMaasVolumeGroupDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMaasVolumeGroupImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"available_size": {
```

Comment:
Available size looks like something dynamic. Maybe it is better to only keep `size`?